### PR TITLE
Improve posix bindings

### PR DESCRIPTION
**Pull request by Korf74**
I  implemented part of the native linux system calls.

### CODE REVIEW

**Comment by Korf74** [comment]
I'll write some tests and commit them sometime in the following days. I'll add the links right now I think

File: nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    extern
+  def getpeername(sockfd: CInt,
+                  addr: Ptr[sockaddr],
+                  addrlen: Ptr[socklen_t]): CInt = extern
+  def send(sockfd: CInt, buf: Ptr[Byte], len: CSize, flags: CInt): ssize_t =
+    extern
+  def sendto(sockfd: CInt,
+             buf: Ptr[Byte],
+             len: CSize,
+             flags: CInt,
```
Comment (MasseGuillaume) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: @densh Do we have a special construct for flags variables?

It would be useful to type this with value classes:

```
class SendToFlags(val underlying: Int) extends AnyVal {
  def &(other: SendToFlags) = new SendToFlags(underlying & other.underlying)
}

abstract class FlagValues[T] {
  private var current = 1
  def next() = {
    val t = current
    current = current << 1
    t
  }
}

object SendToFlags extends FlagValues[SendToFlags] {
  val MSG_CONFIRM = next()
  val MSG_DONTROUTE = next()
  val MSG_DONTWAIT = next()
  val MSG_EOR = next()
  val MSG_MORE = next()
  val MSG_NOSIGNAL = next()
  val MSG_OOB = next()
}

object Main extends App {
	import SendToFlags._

	println(MSG_CONFIRM)
	println(MSG_DONTROUTE)
	println(MSG_DONTWAIT)
	println(MSG_EOR)
	println(MSG_MORE)
	println(MSG_NOSIGNAL)
	println(MSG_OOB)
}

def sendto(sockfd: CInt, buf: Ptr[Byte], len: CSize, flags: SendToFlag, dest_addr: Ptr[sockaddr], addrlen: socklen_t): ssize_t = extern

```

Comment (MasseGuillaume) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: Denys said it should work :-)

Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: We should not alias `+` to mean `|` and `-` to mean `&`. Those options are bitflags and using them should look like bitflags. `+` on bitflags means something completely different and would be extremely confusing for people who are used to posix APIs in other native languages.

Comment (MasseGuillaume) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: sure the question was more can we pass a `class SendToFlags(val underlying: CInt) extends AnyVal` to a facade expecting a `CInt`

Comment (Korf74) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: Should I try to implement this flag handling then ? If so, I should delete the @name macros for flags as well right ?

Comment (MasseGuillaume) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: @Korf74 It's up to you. It can be in a separate PR too.

I think the `@name` macro is to make it available in C. Since it's already available via the header file I think it's redundant and can be removed.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/fsuid.scala [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package scala.scalanative.posix.sys
+
+import scala.scalanative.native.{CInt, extern}
```
Comment (MasseGuillaume) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/fsuid.scala]: you can also break down the imports:

```
package scala.scalanative
package posix.sys

import native.{CInt, extern}
```

File: unit-tests/src/main/scala/scala/scalanative/SyscallsSuite.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import scala.scalanative.native.{CInt, CArray}
+import scala.scalanative.native.Nat.{_2}
+import scala.scalanative.posix.unistd
+import unistd._
+
+/**
+  * Created by remi on 02/03/17.
+  */
+object SyscallsSuite extends tests.Suite {
```
Comment (MasseGuillaume) [inline unit-tests/src/main/scala/scala/scalanative/SyscallsSuite.scala]: https://github.com/MasseGuillaume/scala-native/commit/6ac1e65792334742b7dfe7cdb2672570b3a0a214#diff-83f5d0987d152f874b82a614f4b2b9c5R22

**Comment by densh** [comment]
Hi @Korf74, we're currently having a temporary code freeze before official `0.1.0` announcement. The artefacts have already been published, we can only change the docs for now. This should probably take another week or so to get through.

**Comment by Korf74** [comment]
Ok no problem

**Comment by Korf74** [comment]
Note : because of #555 some functions can't have the exact C format for arguments (e.g pipe should be taking a CArray as argument but takes a Ptr[Carray])

**Comment by Korf74** [comment]
Thank you I changed that !

**Comment by densh** [comment]
@Korf74 This PR conflicts with #574 but we'll probably merge that one first and you'll need to rebase on top of latest master afterwards.

**Comment by densh** [comment]
@Korf74 Needs a rebase on top of latest master.

**Comment by Korf74** [comment]
I just rebased but I'm not sure I did it right it's the first time I do it for such a big project

File: nativelib/src/main/scala/scala/scalanative/runtime/time.scala [hunk 4]
```diff
@@ -5,6 +5,12 @@ import scala.scalanative.native.{CLongLong, extern}
...
 object time {
+<<<<<<< 4459f51eb5b8a0039e00609e5446fea26a349159
   def scalanative_nano_time: CLongLong           = extern
+
   def scalanative_current_time_millis: CLongLong = extern
 }
+=======
+  def scalanative_nano_time: CLongLong = extern
+}
+>>>>>>> second field of timespec is now CLong
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/runtime/time.scala]: You need to resolve this conflict, this is not going to compile.

**Comment by densh** [comment]
@Korf74 It gets a bit easier if you squash everything together into a single commit.

**Comment by MasseGuillaume** [comment]
@Korf74 

To solve this I would do:
`git rebase -i 01c35478914d0b3c2e6cd4f011d5da429a7d67eb`
pick the first one and squash the remaining.
`git commit --amend`
rework the commit message

```
git checkout master
git pull origin master`
git checkout syscalls
git rebase master
```

solve conflicts: 
* `git status`
* solve conflict manually
* `git add` the file where you solved the conflict
* `git rebase --continue` when all conflicts are solved

to solve a conflict you have to be really careful. Make sure you understand the changes you are pulling from. You can look at the master branch to see how the merge should end for example.

**Comment by Korf74** [comment]
@MasseGuillaume I've done the rebase to master already, you want me to squash all my commit to a unique one ?

**Comment by Korf74** [comment]
@densh it seems that the build still fails with a timeout

File: unit-tests/src/main/scala/tests/Main.scala [hunk 5]
```diff
@@ -2,6 +2,8 @@ package tests
 
 import java.lang.System.exit
 
+import scala.scalanative.SyscallsSuite
```
Comment (densh) [inline unit-tests/src/main/scala/tests/Main.scala]: This change is not necessary, it doesn't have any effect. Test suites are resolved automatically.

File: nativelib/src/main/scala/scala/scalanative/native/time.scala [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  def timer_settime(timerid: timer_t,
+                    flags: CInt,
+                    new_value: Ptr[timespec],
+                    old_value: Ptr[timespec]): CInt                    = extern
+  def timer_gettime(timerid: timer_t, curr_value: Ptr[timespec]): CInt = extern
+  def timer_getoverrun(timerid: timer_t): CInt                         = extern
+  def getdate(string: CString): Ptr[tm]                                = extern
+  def getdate_r(string: CString, res: Ptr[tm]): CInt                   = extern
+
+  def scalanative_nano_time: CLongLong = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: This one is neither a POSIX nor C API and should not be here.

File: nativelib/src/main/scala/scala/scalanative/native/time.scala [hunk 7]
```diff
@@ -0,0 +1,84 @@
+package scala.scalanative
+package native
+
+import native.signal.sigevent
+import posix.sys.types.{timer_t, pid_t, clockid_t}
+
+// http://man7.org/linux/man-pages/man7/time.7.html
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: Here we don't aim to do anything linux-specific. Our goal is to do purely bindings for documented POSIX APIs. e.g. http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/time.h.html

File: nativelib/src/main/scala/scala/scalanative/native/time.scala [hunk 8]
```diff
@@ -0,0 +1,84 @@
+package scala.scalanative
+package native
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: `scalanative.native` may only contain bindings to C's standard library definitions. This one seems to be a binding for POSIX time, therefore should go to `scalanative.posix` instead.

File: nativelib/src/main/scala/scala/scalanative/native/time.scala [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  def localtime(timep: Ptr[time_t]): Ptr[tm] = extern
+  def gmtime_r(timep: Ptr[time_t], result: Ptr[tm]): Ptr[tm] =
+    extern
+  def localtime_r(timep: Ptr[time_t], result: Ptr[tm]): Ptr[tm] =
+    extern
+  def asctime(tm: Ptr[tm]): CString                             = extern
+  def asctime_r(tm: Ptr[tm], buf: CString): CString             = extern
+  def ctime(timep: Ptr[time_t]): CString                        = extern
+  def ctime_r(timep: Ptr[time_t], buf: CString): CString        = extern
+  def stime(t: Ptr[time_t]): CInt                               = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: `stime` is not listed in POSIX's `time.h` docs, therefore should not be here.

File: nativelib/src/main/scala/scala/scalanative/native/time.scala [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+  def timegm(tm: Ptr[tm]): time_t                               = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: Not listed in POSIX's `time.h`.

File: nativelib/src/main/scala/scala/scalanative/native/time.scala [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+  def timelocal(tm: Ptr[tm]): time_t                            = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: Not listed in POSIX's `time.h`.

File: nativelib/src/main/scala/scala/scalanative/native/time.scala [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+  def dysize(year: CInt): CInt                                  = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: Not listed in POSIX's `time.h`.

File: nativelib/src/main/scala/scala/scalanative/native/signal.scala [hunk 13]
```diff
@@ -11,6 +13,11 @@ object signal {
     extern
   def raise(sig: CInt): CInt = extern
 
+  // Type
+  type sigval = Unit // Union[2]
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/signal.scala]: This is not a correct binding for 
```
      union sigval {          /* Data passed with notification */
           int     sival_int;         /* Integer value */
           void   *sival_ptr;         /* Pointer value */
       };
```
It should be `Ptr[Byte]`.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+object mman {
+  def mmap(addr: Ptr[Byte],
+           len: CSize,
+           prot: CInt,
+           flags: CInt,
+           fd: CInt,
+           offset: off_t): Ptr[Byte] = extern
+
+  /* Deallocate any mapping for the region starting at ADDR and extending LEN
+   bytes.  Returns 0 if successful, -1 for errors (and sets errno).  */
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala]: We don't include doc comments for bindings as the exact behavior might differ per platform.

File: nativelib/src/main/scala/scala/scalanative/posix/grp.scala [hunk 15]
```diff
@@ -0,0 +1,10 @@
+package scala.scalanative
+package posix
+
+import native.extern
+
+/**
+ * Created by remi on 28/02/17.
+ */
+@extern
+object grp {}
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/grp.scala]: This file is not useful and should be dropped.

File: nativelib/src/main/scala/scala/scalanative/posix/grp.scala [hunk 16]
```diff
@@ -0,0 +1,10 @@
+package scala.scalanative
+package posix
+
+import native.extern
+
+/**
+ * Created by remi on 28/02/17.
+ */
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/grp.scala]: Please remove all authorship comments from the code. We don't use them as they easily get out of date.

File: nativelib/src/main/scala/scala/scalanative/posix/mqueue.scala [hunk 17]
```diff
@@ -0,0 +1,48 @@
+package scala.scalanative
+package posix
+
+import native._
+import native.signal.sigevent
+import native.time.timespec
+import sys.types.{ssize_t, mode_t}
+
+// http://man7.org/linux/man-pages/man7/mq_overview.7.html
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/mqueue.scala]: The same as above for `time.h`, please link to POSIX docs, not linux docs.

File: nativelib/src/main/scala/scala/scalanative/posix/mqueue.scala [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+
+@extern
+object mqueue {
+
+  def mq_open(name: CString,
+              oflag: CInt,
+              mode: mode_t,
+              attr: Ptr[mq_attr]): mqd_t                 = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/mqueue.scala]: This is a binding for linux version, POSIX one differs: http://pubs.opengroup.org/onlinepubs/9699919799/functions/mq_open.html

File: nativelib/src/main/scala/scala/scalanative/posix/sched.scala [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+  /* Get minimum priority value for a scheduler.  */
+  def sched_get_priority_min(algorithm: CInt): CInt = extern
+
+  /* Get the SCHED_RR interval for the named process.  */
+  def sched_rr_get_interval(pid: pid_t, t: Ptr[timespec]): CInt = extern
+
+  /* Set the CPU affinity for a task */
+  def sched_setaffinity(pid: pid_t,
+                        cpusetsize: CSize,
+                        cpuset: Ptr[cpu_set_t]): CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sched.scala]: This one is not listed in POSIX docs.

File: nativelib/src/main/scala/scala/scalanative/posix/sched.scala [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+
+  /* Get the CPU affinity for a task */
+  def sched_getaffinity(pid: pid_t,
+                        cpusetsize: CSize,
+                        cpuset: Ptr[cpu_set_t]): CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sched.scala]: This one is not listed in POSIX docs.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/fsuid.scala [hunk 21]
```diff
@@ -0,0 +1,15 @@
...
+
+// part of http://man7.org/linux/man-pages/man7/credentials.7.html
+
+@extern
+object fsuid {
+
+  def setfsuid(fsuid: uid_t): CInt = extern
+  def setfsgid(fsgid: gid_t): CInt = extern
+
+}
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/fsuid.scala]: This header is not listed in POSIX docs, we should not include it here.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/ipc.scala [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+
+  // Macros
+  @name("scalanative_ipc_stat")
+  def IPC_STAT: CInt = extern
+  @name("scalanative_ipc_set")
+  def IPC_SET: CInt = extern
+  @name("scalanative_ipc_rmid")
+  def IPC_RMID: CInt = extern
+  @name("scalanative_ipc_info")
+  def IPC_INFO: CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/ipc.scala]: This one is not listed in POSIX docs.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/ipc.scala [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+  @name("scalanative_ipc_private")
+  def IPC_PRIVATE: CInt = extern
+  @name("scalanative_ipc_creat")
+  def IPC_CREATE: CInt = extern
+  @name("scalanative_ipc_execl")
+  def IPC_EXECL: CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/ipc.scala]: This one is not listed in posix docs, did you mean IPC_EXCL?

File: nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala [hunk 24]
```diff
@@ -0,0 +1,132 @@
...
+  @name("scalanative_map_noreserve")
+  def MAP_NORESERVE: CInt = extern
+  @name("scalanative_map_populate")
+  def MAP_POPULATE: CInt = extern
+  @name("scalanative_map_stack")
+  def MAP_STACK: CInt = extern
+  @name("scalanative_map_uninitialized")
+  def MAP_UNINITIALIZED: CInt = extern
+  @name("scalanative_config_mmap_allow_uninitialized")
+  def CONFIG_MMAP_ALLOW_UNINITIALIZED: CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala]: Please double check that none of those are linux-specific. 

For example MAP_HUGE_* are not in http://pubs.opengroup.org/onlinepubs/9699919799/

File: nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala [hunk 25]
```diff
@@ -0,0 +1,132 @@
...
+     (and sets errno).  */
+  def mprotect(addr: Ptr[Byte], len: CSize, prot: CInt): CInt = extern
+
+  /* Synchronize the region starting at ADDR and extending LEN bytes with the
+     file it maps.*/
+  def msync(addr: Ptr[Byte], len: CSize, flags: CInt): CInt = extern
+
+  /* Advise the system about particular usage patterns the program follows
+       for the region starting at ADDR and extending LEN bytes.  */
+  def madvise(addr: Ptr[Byte], len: CSize, advice: CInt): CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala]: Another linux-specific API.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala [hunk 26]
```diff
@@ -0,0 +1,132 @@
+package scala.scalanative
+package posix.sys
+
+import native._
+import types.{mode_t, off_t}
+
+@extern
+object mman {
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/mman.scala]: This bindings are missing some APIs with posix_ prefix as defined in http://pubs.opengroup.org/onlinepubs/9699919799/

File: nativelib/src/main/scala/scala/scalanative/posix/sys/msg.scala [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
+  @name("scalanative_msg_info")
+  def MSG_INFO: CInt = extern
+  @name("scalanative_msg_stat")
+  def MSG_STAT: CInt = extern
+  @name("scalanative_msg_noerror")
+  def MSG_NOERROR: CInt = extern
+  @name("scalanative_msg_copy")
+  def MSG_COPY: CInt = extern
+  @name("scalanative_msg_except")
+  def MSG_EXCEPT: CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/msg.scala]: Out of those only `MSG_NOERROR` is listed in http://pubs.opengroup.org/onlinepubs/9699919799/

File: nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala [hunk 28]
```diff
@@ -0,0 +1,160 @@
+package scala.scalanative
+package posix.sys
+
+import native.Nat._
+import native._
+import native.time.timespec
+import types.ssize_t
+import uio.iovec
+
+// http://man7.org/linux/man-pages/man7/socket.7.html
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: http://pubs.opengroup.org/onlinepubs/9699919799/

File: nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala [hunk 29]
```diff
@@ -0,0 +1,160 @@
...
+             buf: Ptr[Byte],
+             len: CSize,
+             flags: CInt,
+             dest_addr: Ptr[sockaddr],
+             addrlen: socklen_t): ssize_t                           = extern
+  def sendmsg(sockfd: CInt, msg: Ptr[msghdr], flags: CInt): ssize_t = extern
+  def sendmmsg(sockfd: CInt,
+               msgvec: Ptr[mmsghdr],
+               vlen: CUnsignedInt,
+               flags: CUnsignedInt): CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: Not listed in POSIX APIs.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala [hunk 30]
```diff
@@ -0,0 +1,160 @@
...
+               len: CSize,
+               flags: CInt,
+               src_addr: Ptr[sockaddr],
+               addrlen: socklen_t): ssize_t                         = extern
+  def recvmsg(sockfd: CInt, msg: Ptr[msghdr], flags: CInt): ssize_t = extern
+  def recvmmsg(sockfd: CInt,
+               msgvec: Ptr[mmsghdr],
+               vlen: CUnsignedInt,
+               flags: CUnsignedInt,
+               timeout: Ptr[timespec]): CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: Not listed in POSIX APIs.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala [hunk 31]
```diff
@@ -0,0 +1,160 @@
...
+                 optname: CInt,
+                 optval: Ptr[Byte],
+                 optlen: socklen_t): CInt       = extern
+  def listen(sockfd: CInt, backlog: CInt): CInt = extern
+  def accept(sockfd: CInt, addr: Ptr[sockaddr], addlen: Ptr[socklen_t]): CInt =
+    extern
+  def accept4(sockfd: CInt,
+              addr: Ptr[sockaddr],
+              addlen: Ptr[socklen_t],
+              flags: CInt): CInt              = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: Not listed in POSIX APIs.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala [hunk 32]
```diff
@@ -0,0 +1,160 @@
...
+  @name("scalanative_msg_peek")
+  def MSG_PEEK: CInt = extern
+  @name("scalanative_msg_trunc")
+  def MSG_TRUNC: CInt = extern
+  @name("scalanative_msg_waitall")
+  def MSG_WAITALL: CInt = extern
+  @name("scalanative_msg_ctrunc")
+  def MSG_CTRUNC: CInt = extern
+  @name("scalanative_msg_waitforone")
+  def MSG_WAITFORONE: CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala]: Please double check that none of the above include linux-specific things.

File: unit-tests/src/main/scala/scala/scalanative/SyscallsSuite.scala [hunk 33]
```diff
@@ -0,0 +1,63 @@
+package scala.scalanative
+
+import native._
+import posix.unistd
+import scala.scalanative.native.Nat._2
+
+object SyscallsSuite extends tests.Suite {
+
+  test("pipe + dup + getpid + getppid + fork + read + write") {
```
Comment (densh) [inline unit-tests/src/main/scala/scala/scalanative/SyscallsSuite.scala]: Tests should be more finer-grain, please split into smaller ones that isolate minimum number of calls at a time.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/timex.scala [hunk 34]
```diff
@@ -0,0 +1,108 @@
+package scala.scalanative
+package posix.sys
+
+import native._
+import types.suseconds_t
+import native.time.time_t
+
+@extern
+object timex {
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/timex.scala]: This is not a POSIX API.

File: nativelib/src/main/scala/scala/scalanative/posix/sys/uio.scala [hunk 35]
```diff
@@ -0,0 +1,12 @@
...
+package posix.sys
+
+import native.{CSize, Ptr, CStruct2, extern}
+
+@extern
+object uio {
+
+  // Types
+  type iovec = CStruct2[Ptr[Byte], CSize]
+
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/posix/sys/uio.scala]: `uio.h` contains two functions: `readv` and `writev`. You should include them here.

http://pubs.opengroup.org/onlinepubs/9699919799/

**Comment by densh** [comment]
@Korf74 Sorry on the delay of the review for this PR. In the future please open smaller finer-grain PRs, this will make sure that every single change is mergeable quickly. It's also really easy to miss something when you get to review 1000+ LOC diff.

**Comment by Korf74** [comment]
@densh Thanks for the pass I'll tend to the changes asap, might take a couple of weeks I'm a bit in a rush these days

**Comment by densh** [comment]
@Korf74 It might be easier to close this PR and open smaller ones, one header at a time. This way there is no pressure to do lots of changes at once. What do you think?

**Comment by Korf74** [comment]
@densh I can do it indeed but won't it pollute the PRs with x of them being header files ?

**Comment by densh** [comment]
As long as you have a few of them open (lets say 2-3) at a time -- it's not a problem at all. It will make it much easier to review them too.

**Comment by Korf74** [comment]
@densh ok then

**Comment by densh** [comment]
@Korf74 I'm closing this one, looking forward to smaller, finer-grain PRs! :)
